Add product description title in old product editor

louwie17 opened the pull request:
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

Added a product description title to the product text editor, using a similar format as the short description, making use of the  postbox styling.
There wasn't an overly clean way to do this aside from doing it through JavaScript, as the `edit-form-advanced.php` file doesn't provide many filters to wrap the `wp_editor` in: https://github.com/WordPress/WordPress/blob/master/wp-admin/edit-form-advanced.php#L581-L649
The only options would be to make use of the `edit_form_after_title` and remove the `post_type_supports`, but this could break backwards compatibility and the `edit_form_after_title` does not seem like the right action to use for this.

One side note to my JavaScript implementation, I decided to only append the title html and add a class to handle the remaining styling. I did try to actually wrap the `editor` html using `innerHtml`, but this seemed to break the bootstrapping of the editor.

<img width="1373" alt="Screen Shot 2022-10-21 at 8 59 16 AM" src="https://user-images.githubusercontent.com/2240960/197190802-a10408fd-f142-4360-a99c-f01b78914077.png">
<img width="1371" alt="Screen Shot 2022-10-21 at 8 59 05 AM" src="https://user-images.githubusercontent.com/2240960/197190805-d1d91f34-aa2a-4c29-a268-44cc788fd6a1.png">

cc @jarekmorawski for styling approval, see above two images.

Closes #33538  .

### How to test the changes in this Pull Request:

1. Load this branch and go to **Products > Add New** and notice how the long description is wrapped in a panel named **Product Description**
2. Do the same check for editing an existing product
3. Now download the **Classic Editor** plugin and edit an existing post or page and make sure they still look the same.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> run changelog add`?

<!-- Mark completed items with an [x] -->

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

louwie17 committed:
Add product description title to classic product editor

louwie17 committed:
Remove bottom margin as its already added by the parent element

louwie17 committed:
Add changelog

joshuatf commented on plugins/woocommerce/client/legacy/css/admin.scss:
```diff
@@ -7780,3 +7780,20 @@ table.bar_chart {
+		margin: 6px 12px 0;
+	}
+	#post-status-info {
+		margin: 0px 12px 12px;
```
Do we have access to our usual scss vars in here?  If so we could use gap vars like `$gap-small`.

louwie17 replied:
Unfortunately it doesn't seem like we do.

joshuatf commented on plugins/woocommerce/client/legacy/js/admin/product-editor.js:
```diff
@@ -0,0 +1,15 @@
+/* global woocommerce_admin_product_editor */
+jQuery( function ( $ ) {
```
jQuery 💥

joshuatf commented on plugins/woocommerce/includes/admin/class-wc-admin-assets.php:
```diff
@@ -233,6 +233,15 @@ public function admin_scripts() {
+				wp_enqueue_script( 'wc-admin-product-editor', WC()->plugin_url() . '/assets/js/admin/product-editor' . $suffix . '.js', array( 'jquery' ), $version );
+
+				wp_localize_script( 'wc-admin-product-editor', 'woocommerce_admin_product_editor', array(
+					'i18n_description' => esc_js( __( 'Product Description', 'woocommerce' ) ),
```
I think this should be a lowercase "D" to match the other field titles on the page:

```suggestion
					'i18n_description' => esc_js( __( 'Product description', 'woocommerce' ) ),
```

joshuatf reviewed (dismissed):
Nice work!  Flashbacks to the old days of jQuery.

Left one question and a minor clean-up comment, but pre-approving as everything is working as described.

The design feels slightly odd and eats into valuable screen real estate on smaller screens, but I'll leave this up to @jarekmorawski to approve.
<img width="401" alt="Screen Shot 2022-10-19 at 4 08 54 PM" src="https://user-images.githubusercontent.com/10561050/196821439-e80fbcfa-3fac-44ad-9c5e-4ba4ccd5773c.png">

jarekmorawski commented:
> cc @jarekmorawski for styling approval, see above two images.

Can we add a bottom padding to the permalink element, so the space between the product name input field and the description card is maintained? It looks off when they're this close to each other.

> The design feels slightly odd and eats into valuable screen real estate on smaller screens, but I'll leave this up to @jarekmorawski to approve.

What about it feels odd to you, Josh? It does take up more space, but not unlike all other cards on the screen. I would keep it as is.

louwie17 committed:
Update css and name, also fix some lint errors

louwie17 committed:
Fix spacing

louwie17 commented:
Thanks @jarekmorawski I updated the screenshots above with the new spacing.

> Can we add a bottom padding to the permalink element, so the space between the product name input field and the description card is maintained? It looks off when they're this close to each other.

Instead of add a bottom padding to the permalink, I added a margin to the top of the description meta box. Mostly because this would prevent the Product description to jump down once the permalink becomes visible. This is how the previous design was laid out as well. The extra spacing doesn't appear to be to noticable.

louwie17 commented:
@joshuatf this should be ready for a re-review

jarekmorawski commented:
I agree. It looks good to me!

joshuatf reviewed (approved):
Sorry about the late re-review!  LGTM 🚢 

>What about it feels odd to you, Josh? It does take up more space, but not unlike all other cards on the screen. I would keep it as is.

I suppose because unlike other cards it's an editing experience and on smaller screens could reduce the editor by almost 10%.  But if this looks good to you, then it's fine by me.  I also think the number of users editing on smaller screens is relatively small.

jarekmorawski commented:
> I suppose because unlike other cards it's an editing experience and on smaller screens could reduce the editor by almost 10%. But if this looks good to you, then it's fine by me. I also think the number of users editing on smaller screens is relatively small.

AFAIK it's in single digits. I don't think we need to focus on it too much now. We'll do mobile right in the new UX. 👍
